packages/effect/src/DateTime.ts

```diff
@@ -0,0 +1,1991 @@
+
+/**
+ * Add the given `Duration` to a `DateTime`.
+ *
+ * @since 3.6.0
+ * @category math
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // add 5 minutes
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.addDuration("5 minutes")
+ * )
+ */
+export const addDuration: {
+  (duration: Duration.DurationInput): <A extends DateTime>(self: A) => DateTime.PreserveZone<A>
+  <A extends DateTime>(self: A, duration: Duration.DurationInput): DateTime.PreserveZone<A>
+} = dual(
+  2,
+  (self: DateTime, duration: Duration.DurationInput): DateTime =>
+    mapEpochMillis(self, (millis) => millis + Duration.toMillis(duration))
+)
+
+/**
+ * Subtract the given `Duration` from a `DateTime`.
+ *
+ * @since 3.6.0
+ * @category math
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // subtract 5 minutes
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.subtractDuration("5 minutes")
+ * )
+ */
+export const subtractDuration: {
+  (duration: Duration.DurationInput): <A extends DateTime>(self: A) => DateTime.PreserveZone<A>
+  <A extends DateTime>(self: A, duration: Duration.DurationInput): DateTime.PreserveZone<A>
+} = dual(
+  2,
+  (self: DateTime, duration: Duration.DurationInput): DateTime =>
+    mapEpochMillis(self, (millis) => millis - Duration.toMillis(duration))
+)
+
+const addMillis = (date: DateTime, amount: number): DateTime => mapEpochMillis(date, (millis) => millis + amount)
+
+/**
+ * Add the given `amount` of `unit`'s to a `DateTime`.
+ *
+ * The time zone is taken into account when adding days, weeks, months, and
+ * years.
+ *
+ * @since 3.6.0
+ * @category math
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // add 5 minutes
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.add(5, "minutes")
+ * )
+ */
+export const add: {
+  (amount: number, unit: DateTime.Unit): <A extends DateTime>(self: A) => DateTime.PreserveZone<A>
+  <A extends DateTime>(self: A, amount: number, unit: DateTime.Unit): DateTime.PreserveZone<A>
+} = dual(3, (self: DateTime, amount: number, unit: DateTime.Unit): DateTime => {
+  switch (unit) {
+    case "millis":
+    case "milli":
+      return addMillis(self, amount)
+    case "seconds":
+    case "second":
+      return addMillis(self, amount * 1000)
+    case "minutes":
+    case "minute":
+      return addMillis(self, amount * 60 * 1000)
+    case "hours":
+    case "hour":
+      return addMillis(self, amount * 60 * 60 * 1000)
+  }
+  return mutate(self, (date) => {
+    switch (unit) {
+      case "days":
+      case "day": {
+        date.setUTCDate(date.getUTCDate() + amount)
+        return date
+      }
+      case "weeks":
+      case "week": {
+        date.setUTCDate(date.getUTCDate() + amount * 7)
+        return date
+      }
+      case "months":
+      case "month": {
+        const day = date.getUTCDate()
+        date.setUTCMonth(date.getUTCMonth() + amount + 1, 0)
+        if (day < date.getUTCDate()) {
+          date.setUTCDate(day)
+        }
+        return date
```

Comment:
This aligns with `Temporal`'s default `{ overflow: "constrain" }` option, but it's different to how standard `Date`s work. Just want to point it out to make sure it's intentional and that there's also no intention to add another option for it.

Reply:
Yes it is intentional, I think it is temporals default for good reason.

If you want the more exact behaviour, you can use .addDuration.